File: training-data/go/priority_queue.go
package main

import (
	"container/heap"
	"fmt"
)

type pqItem struct {
	value    string
	priority int
}

type priorityQueue []*pqItem

func (pq priorityQueue) Len() int            { return len(pq) }
func (pq priorityQueue) Less(i, j int) bool  { return pq[i].priority < pq[j].priority }
func (pq priorityQueue) Swap(i, j int)       { pq[i], pq[j] = pq[j], pq[i] }
func (pq *priorityQueue) Push(x interface{}) { *pq = append(*pq, x.(*pqItem)) }
func (pq *priorityQueue) Pop() interface{} {
	old := *pq
	n := len(old)
	item := old[n-1]
	*pq = old[:n-1]
	return item
}

func main() {
	pq := &priorityQueue{}
	heap.Init(pq)
	heap.Push(pq, &pqItem{"low", 5})
	heap.Push(pq, &pqItem{"high", 1})
	heap.Push(pq, &pqItem{"medium", 3})
	for pq.Len() > 0 {
		fmt.Println(heap.Pop(pq).(*pqItem).value)
	}
}

File: training-data/python/priority_queue.py
import heapq
from dataclasses import dataclass, field
from typing import Any


@dataclass(order=True)
class PrioritizedItem:
    priority: int
    item: Any = field(compare=False)


class PriorityQueue:
    def __init__(self):
        self._heap = []

    def push(self, item, priority: int):
        heapq.heappush(self._heap, PrioritizedItem(priority, item))

    def pop(self):
        return heapq.heappop(self._heap).item

    def is_empty(self):
        return not self._heap


if __name__ == "__main__":
    pq = PriorityQueue()
    pq.push("low", 5)
    pq.push("high", 1)
    pq.push("medium", 3)
    print(pq.pop(), pq.pop(), pq.pop())
